routing.md

```diff
@@ -1,4 +1,275 @@
+
+### Two Dimensions
+
+Two dimensions of routing are relevant to DIDComm. Sometimes they are confused or conflated:
+
+* __Network routing__ deals with how packets flow across a digital landscape. This is the "routing" familiar to most technical people, and the one that will inform most assumptions they bring to the routing topic when reading the DIDComm spec for the first time.
+* __Cryptographic routing__ concerns itself with how packaging hides or reveals plaintext to participants in a delivery chain. In other words, it's about the routing of secrets rather than the routing of packets.
+
+The overview immediately above collapsed these two dimensions, assuming that a node relaying data is also a node decrypting. This makes for simple explanations, but it's not always a true or helpful assumption. The correspondence between these two routing dimensions can be non-trivial; a route may have ten network hops but only three cryptographic hops, and a single network hop may decrypt twice as it passes data between distinct software entities. However, simple principles explain the dynamics for each situation.
+
+### Mediators and Relays
+
+DIDComm routing uses two important constructs to model all varieties of network and cryptographic routing: _mediators_ and _relays_.
+
+A __mediator__ is a participant in routing that must be accounted for by the sender's cryptography. In other words, it is visible in the cryptographic routing dimension. It has its own keys and will deliver messages only after decrypting an outer envelope to reveal a `forward` request. It must understand DIDComm routing to do this. Many types of mediators may exist, but two important ones should be widely understood, as they commonly manifest in DID Docs:
+
+1. A service that hosts many agents at a single endpoint to provide herd privacy (sometimes called an "agency") is a _mediator_.
```

Comment:
I think `hosts` may have unintended associations. I propose `receives messages for` or `receives messages on behalf of` as a replacement.

Reply:
Okay. I will update.

Reply:
Fixed